vortex-scalar/src/binary.rs

```diff
@@ -52,47 +52,58 @@ impl<'a> TryFrom<&'a Scalar> for Buffer {
     type Error = VortexError;
 
     fn try_from(scalar: &'a Scalar) -> VortexResult<Self> {
-        Buffer::try_from(scalar.value())
+        <Option<Buffer>>::try_from(scalar)?
+            .ok_or_else(|| vortex_err!("Can't extract present value from null scalar"))
     }
 }
 
-impl TryFrom<Scalar> for Buffer {
+impl<'a> TryFrom<&'a Scalar> for Option<Buffer> {
     type Error = VortexError;
 
-    fn try_from(scalar: Scalar) -> VortexResult<Self> {
-        Buffer::try_from(&scalar)
+    fn try_from(scalar: &'a Scalar) -> VortexResult<Self> {
+        Ok(BinaryScalar::try_from(scalar)?.value())
     }
 }
 
-impl TryFrom<&ScalarValue> for Buffer {
+impl TryFrom<Scalar> for Buffer {
     type Error = VortexError;
 
-    fn try_from(value: &ScalarValue) -> Result<Self, Self::Error> {
-        Option::<Buffer>::try_from(value)?
-            .ok_or_else(|| vortex_err!("Can't extract present value from null scalar"))
+    fn try_from(scalar: Scalar) -> VortexResult<Self> {
+        Self::try_from(&scalar)
     }
 }
 
-impl TryFrom<ScalarValue> for Buffer {
+impl TryFrom<Scalar> for Option<Buffer> {
     type Error = VortexError;
 
-    fn try_from(value: ScalarValue) -> Result<Self, Self::Error> {
-        Buffer::try_from(&value)
+    fn try_from(scalar: Scalar) -> VortexResult<Self> {
+        Self::try_from(&scalar)
     }
 }
 
-impl TryFrom<&ScalarValue> for Option<Buffer> {
-    type Error = VortexError;
-
-    fn try_from(value: &ScalarValue) -> Result<Self, Self::Error> {
-        value.as_buffer()
+impl From<&[u8]> for Scalar {
+    fn from(value: &[u8]) -> Self {
+        Self {
+            dtype: DType::Binary(Nullability::NonNullable),
+            value: ScalarValue(InnerScalarValue::Buffer(value.into())),
+        }
     }
 }
 
-impl TryFrom<ScalarValue> for Option<Buffer> {
-    type Error = VortexError;
+impl From<bytes::Bytes> for Scalar {
```

Comment:
could we implement `From<Into<Buffer>>` or some such?

Reply:
leaving off here; all above should be resolved.